tt_metal/common/bfloat8.hpp

```diff
@@ -124,10 +126,13 @@ inline std::vector<float> unpack_bfp8_tiles_into_float_vec(
     auto subtiles_in_tile_col = tile_W / face_W;
     auto subtile_rows = face_H;
     auto subtile_cols = face_W;
-    uint32_t num_exp_words = num_faces * face_H / 4;
+    uint32_t num_exp_words = tt::round_up(num_faces * face_H, l1_alignment) / 4;
     uint32_t num_tile_words = tile_HW / 4;
     uint32_t num_bfp8_in_tile = num_tile_words + num_exp_words;
 
+    // mask out all the extra bits when calculating the exponent index in the exponent dword
```

Comment:
Please add another line, something like:
```
the exponent index will always be 0 when tile_HW == 16, between 0-1 when tile_HW == 32, and between 0-3 otherwise
```